R/pretty_lollipop_plot.R

```diff
@@ -2,58 +2,118 @@
+    maf_data <- read.maf(maf_data)
   }
+  
+  maf_data <- maf_data@data 
+  # Filter for the specific gene 
+  gene_df <- maf_data %>% 
+    filter(Hugo_Symbol == gene_symbol) %>%
+    mutate(AA = as.numeric(gsub("[^0-9]+", "", gsub("([0-9]+).*", "\\1", HGVSp_Short)))) %>%
+    arrange(AA)
 
-  if(missing(plot_title)){
-    plot_title = gene
+  gene_counts <- gene_df %>% 
+    group_by(AA, Start_Position, 
+             End_Position, 
+             Variant_Classification, 
+             Reference_Allele, 
+             Tumor_Seq_Allele2) %>%
+    arrange(AA) %>%
+    summarise(mutation_count = n()) 
+
+  protein_domain_subset <- subset(protein_domains, HGNC == gene_symbol)
+  if (!is.data.frame(protein_domain_subset)) {
+    protein_domain_subset <- as.data.frame(protein_domain_subset)
   }
+   
+  domain_data <- protein_domain_subset %>% data.frame(start.points = protein_domain_subset$Start,
+                                                      end.points = protein_domain_subset$End,
+                                                      text.label = protein_domain_subset$Label,
+                                                      color = protein_domain_subset$Label)
+
+  domain_data$text.position <- (domain_data$start.points + domain_data$end.points) / 2
+
+  # Determine the x-axis range
+  x_max <- max(max(domain_data$end.points), 
+               max(gene_counts$AA, na.rm = TRUE))
+  x_min <- 0
+
+  colors_manual <- c(setNames(protein_domain_subset$Label, protein_domain_subset$Label), 
+                     "Frame_Shift_Del" = "#e50505", 
+                     "Frame_Shift_Ins" = "#df9307" ,
+                     "In_Frame_Del" = "#f6f60c",
+                     "In_Frame_Ins" = "#04e504",
+                     "Missense_Mutation" = "#9e12f6", 
+                     "Nonsense_Mutation" = "#06ebeb",
+                     "Nonstop_Mutation" = "#0505df", 
+                     "Splice_Site" = "#035703",
+                     "Translation_Start_Site" = "#f3899b",
+                     "RNA" = "#146deb")
+  
+  Somatic_Mutation_Statistic_Numerator <- maf_data %>%
+    filter(Hugo_Symbol == gene_symbol) %>% 
+    filter(!(Variant_Classification %in% c("Intron",
+                                          "Silent",
+                                          "3'UTR",
+                                          "5'UTR",
+                                          "IGR",
+                                          "3'Flank",
+                                          "5'Flank",
+                                          "Splice_Region"))) %>% 
+    distinct(Tumor_Sample_Barcode) %>% 
+    nrow()
+  
+  Somatic_Mutation_Statistic_Denominator <- length(unique(maf_data$Tumor_Sample_Barcode))
+
+  Somatic_Mutation_Rate <- Somatic_Mutation_Statistic_Numerator/Somatic_Mutation_Statistic_Denominator *100
+  Somatic_Mutation_Rate <- round(Somatic_Mutation_Rate, 2)
 
-  maf_df = maf_df %>%
-    dplyr::filter(Hugo_Symbol == gene)
+  plot <- ggplot() +
```

Comment:
Here we can also switch to a top-to-bottom style instead of left-to-right to adhere to best practices. The example of the gglot call: https://github.com/morinlab/GAMBLR.viz/blob/f12f9641d5f6018e3ac7305ea53a68fcabbca2d5/R/prettyRainfallPlot.R#L482-L514